src/Microsoft.DocAsCode.Build.Engine/Incrementals/BuildMessageInfo.cs

```diff
@@ -3,18 +3,46 @@
 
 namespace Microsoft.DocAsCode.Build.Engine.Incrementals
 {
+    using System;
+    using System.Collections.Concurrent;
     using System.Collections.Generic;
     using System.Linq;
 
     using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.Utility;
 
     public sealed class BuildMessageInfo
     {
         private Listener _listener;
+        private readonly ConcurrentDictionary<string, List<LogItem>> _logs;
```

Comment:
I check the Logger class, there is a lock in Logger.cs at 64, so no need to handle thready safety here.